File: GeneratedEntityReaderClasses/Components/SpriteStainsComponent.cs
namespace GeneratedNoitaClasses;
public class SpriteStainsComponent : NoitaComponentBase,  iNoitaType<SpriteStainsComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaInt sprite_id { get; set; } = new();
	public NoitaBool fade_stains_towards_srite_top { get; set; } = new();
	public NoitaLensValue<NoitaInt> stain_shaken_drop_chance_multiplier { get; set; } = new();
	

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		sprite_id.Read(s);
		fade_stains_towards_srite_top.Read(s);
		stain_shaken_drop_chance_multiplier.Read(s);
		

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		sprite_id.Write(s);
		fade_stains_towards_srite_top.Write(s);
		stain_shaken_drop_chance_multiplier.Write(s);
		

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/SpriteStainsComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaInt
from BasicNoitaTypes import NoitaLensValue
from BasicNoitaTypes import NoitaInt

from NoitaStream import NoitaStream
@dataclass
class SpriteStainsComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    sprite_id:NoitaInt = field(default_factory=NoitaInt)
    fade_stains_towards_srite_top:NoitaBool = field(default_factory=NoitaBool)
    stain_shaken_drop_chance_multiplier:NoitaLensValue = field(default_factory=lambda: NoitaLensValue(NoitaInt))
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.sprite_id.Read(s)
        self.fade_stains_towards_srite_top.Read(s)
        self.stain_shaken_drop_chance_multiplier.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.sprite_id.Write(s)
        self.fade_stains_towards_srite_top.Write(s)
        self.stain_shaken_drop_chance_multiplier.Write(s)

